builtin/providers/cloudstack/resource_cloudstack_network.go

```diff
@@ -166,6 +196,8 @@ func resourceCloudStackNetworkRead(d *schema.ResourceData, meta interface{}) err
 	d.Set("name", n.Name)
 	d.Set("display_text", n.Displaytext)
 	d.Set("cidr", n.Cidr)
+	d.Set("gateway", n.Gateway)
+	d.Set("vlan", n.Vlan)
```

Comment:
I've been playing around and you're right, domain admins can't set or see the vlan:

```
> createNetwork name=Guest92 zoneid=82ad7f4d-262b-4144-8919-e733cf33278d networkofferingid=f994a702-1525-4e0a-8472-ed65ca470e07 displaytext=Guest92 vlan=92
Request Header Fields Too Large: Only ROOT admin is allowed to specify vlanId
```

So reading it back is out of the question (also, it can't be modified without destroying the network so it doesn't matter). However, I still would like seeing the feature there. We use cloudstack as an internal product so we use these vlans extensively. I'll update the docs to put a similar message to the cloudstack's api.